src/core/server/plugins/discovery/plugins_discovery.ts

```diff
@@ -93,7 +93,7 @@ function processPluginSearchPaths$(pluginDirs: readonly string[], log: Logger) {
             // Filter out non-directory entries from target directories, it's expected that
             // these directories may contain files (e.g. `README.md` or `package.json`).
             // We shouldn't silently ignore the entries we couldn't get stat for though.
-            mergeMap(pathStat => (pathStat.isDirectory() ? [path] : [])),
+            mergeMap(pathStat => ((pathStat as Stats).isDirectory() ? [path] : [])),
```

Comment:
What caused this change? `fsStat$(path)` does not longer returns an exact `Stats` type?

Reply:
I have to admit that I'm not very good with `rxjs`, but the problem was the `pathStat` type could not be inferred and was of type `unknown` in the `mergeMap` callback:

![image](https://user-images.githubusercontent.com/10602/78643666-e3d42900-78b4-11ea-8e06-88f6ff5c629c.png)

And so calling `isDirectory()` on it would make TypeScript complain with the following error:

```
Object is of type 'unknown'.
```

Why this wasn't also an issue in Node.js 10 is a mystery to me. In Node.js 10 TypeScript correctly identifies `pathStat` as being of type `Stats`:

![image](https://user-images.githubusercontent.com/10602/78643326-64465a00-78b4-11ea-81fc-64cb16d6331c.png)

**If anyone knows why there's a difference here and if it can/should be fixed elsewhere, I'm all ears 🙂**

Side-note: I would normally have written this line as:

```diff
- mergeMap(pathStat => ((pathStat as Stats).isDirectory() ? [path] : [])),
+ mergeMap((pathStat: Stats) => (pathStat.isDirectory() ? [path] : [])),
```

But this is apparently not allowed with `rxjs` 🤷‍♂

Reply:
This will do for now.